src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,123 @@
+import task.Event;
+import task.Task;
+import task.Todo;
+
+/**
+ * Handle the read and storage of data.
+ */
+public class Storage {
+    private String pathName;
+
+    public Storage(String pathName) {
+        this.pathName = pathName;
+    }
+
+    /**
+     * Convert the time from string to a LocalDate datatype.
+     * 
+     * @param time
+     * @return a LocalDate saving the date information.
+     */
+    public LocalDate readAsDate(String time) {
+        String[] parts = time.split(" ");
+        String datePart = parts[0];
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        LocalDate localDate = LocalDate.parse(datePart, formatter);
+        return localDate;
+    }
+
+    public void autoUpdate(ArrayList<Task> todoList) {
+        File file = new File(this.pathName);
+        file.getParentFile().mkdirs();
+        try {
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            BufferedWriter writer = new BufferedWriter(new FileWriter(file));
+            for (Task t : todoList) {
+                if (t instanceof Todo) {
+                    writer.write("T | " + t.getIsDone() + " | " + t.getDescription());
+                } else if (t instanceof Deadline) {
+                    Deadline d = (Deadline) t;
+                    writer.write("D | " + d.getIsDone() + " | " + d.getDescription() + " | " + d.getBy());
+                } else if (t instanceof Event) {
+                    Event e = (Event) t;
+                    writer.write("E | " + e.getIsDone() + " | " + e.getDescription() + " | " + e.getFrom() + " | "
+                            + e.getTo());
+                }
+                writer.newLine();
+            }
+            writer.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public ArrayList<Task> getHistory() {
+        String fileName = "./data/duke.txt";
+        ArrayList<Task> historyList = new ArrayList<>();
+        File file = new File(fileName);
+        if (!file.exists()) {
+            return new ArrayList<Task>();
+        }
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Task t = createTaskFromLine(line);
+                if (t != null) {
+                    historyList.add(t);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return historyList;
+    }
+
+    private Task createTaskFromLine(String line) {
+        String[] parts = line.split(" \\| ");
+        if (parts.length >= 3) {
+            boolean isDone = Boolean.parseBoolean(parts[1]);
+            String description = parts[2];
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+            switch (parts[0]) {
+                case "T":
```

Comment:
Try to use an enum instead of string matching

Reply:
Noted with thank!